dandi/files/zarr.py

```diff
@@ -209,32 +216,44 @@ def get_validation_errors(
         import zarr
 
         errors: list[ValidationResult] = []
+        origin: Origin = Origin(
+            validator=Validator.zarr,
+            validator_version=zarr.__version__,
+            standard=Standard.ZARR,
+            # TODO: standard_version=...,
+        )
+
         try:
             data = zarr.open(str(self.filepath))
-        except Exception:
+        except Exception as e:
             if devel_debug:
                 raise
             errors.append(
                 ValidationResult(
-                    origin=ValidationOrigin(
-                        name="zarr",
-                        version=zarr.version.version,
-                    ),
+                    origin=origin,
                     severity=Severity.ERROR,
                     id="zarr.cannot_open",
                     scope=Scope.FILE,
+                    origin_result=e,
                     path=self.filepath,
                     message="Error opening file.",
                 )
             )
             data = None
+
+        origin = Origin(
+            validator=Validator.dandi_zarr,
+            validator_version=__version__,
+            standard=Standard.ZARR,
+            # TODO: standard_version=...,
```

Comment:
Per https://zarr.readthedocs.io/en/stable/user-guide/v3_migration.html#migration-guide, it seems that the Python zarr pacakge can support more than one Zarr format 3 specification at the same time. What is the way to determine which Zarr format specification is being used in just opening a file? My bet is that possibly all support format specifications are used, so the standard_version field may be irrelevant and should be skipped in this case.